docs/sources/clients/promtail/stages/logfmt.md

```diff
@@ -0,0 +1,43 @@
+---
+title: logfmt
+---
+# `logfmt` stage
+
+The `logfmt` stage is a parsing stage that reads the log line as [logfmt](https://www.brandur.org/logfmt).
+
+## Schema
+
+```yaml
+logfmt:
+  # Maps set of keys/values extracted from the data
+  #
+  mapping:
+    [ <string>: <string> ... ]
+```
+
+This stage uses the go-kit/log structured logging parser to extract key/value pairs of data from log lines into the extracted map.
+
+## Example
+
+For the given pipeline:
+
+```yaml
+- logfmt:
+    mapping:
+      level: level
+      tag: tag
+      module: module
```

Comment:
the value part can be empty if you do not want to have data extracted with a different name i.e you can have it configured as below:
```
      level:
      tag:
      module:
```

I think it would be a good idea to club both scenarios in the example like
```
      level:
      tag:
      source: module
```
This would extract the following data:
```
- `level`: `info`
- `tag`: `stopping_fetchers`
- `source`: `kafka.consumer.ConsumerFetcherManager`
```